### ISSUE TITLE
[BUG] Expo Support - disableDeviceCollection 

### ISSUE THREAD

**Issue by poikjo** [microsoft/applicationinsights-react-native#28]
**Description/Screenshot**
"Disabling automatic device info collection" documentation says disableDeviceCollection needs to be initialized on ApplicationInsights config. However, ApplicationInsights config does not recognize this property. 

If disableDeviceCollection is initialized on ReactNativeManualDevicePlugin the Android bundling tasks still fails on exception "Unable to resolve 'react-native-device-info'"

If then installing react-native-device-info' you'll end up to exception "NativeModule.RNDeviceInfo is null".

**Steps to Reproduce**

 - React-Native Version: 0.72.3
 - SDK Version [e.g. 22]: 
     "@microsoft/applicationinsights-react-native": "^4.0.0",
    "@microsoft/applicationinsights-web": "^3.0.2",
 - How you initialized the SDK:
`
const RNMPlugin = new ReactNativeManualDevicePlugin({
  disableDeviceCollection: true,
})
const appInsights = new ApplicationInsights({
  config: {
    instrumentationKey: 'YOUR_INSTRUMENTATION_KEY_GOES_HERE',
    extensions: [RNMPlugin],
  },
})
appInsights.loadAppInsights()
`

**Expected behavior**
Device info collection would be ignored

**Additional context**
This might be just documentation issue which needs to be clarified more.

**Comment by MSNev** [comment]
Please see the correct way to pass in the configuration https://github.com/microsoft/applicationinsights-react-native#disabling-automatic-device-info-collection

The signature of the class is currently incorrectly including a "config" during the constructor of the class.

```ts
const RNMPlugin = new ReactNativeManualDevicePlugin();
const appInsights = new ApplicationInsights({   
    config: { 
        instrumentationKey: 'YOUR_INSTRUMENTATION_KEY_GOES_HERE',
        disableDeviceCollection: true,    // <----- The configuration must be passed into the ApplicationInsights config
        extensions: [RNMPlugin]
    }
});
appInsights.loadAppInsights()
```

It may also be included in the extensionConfig

```ts
const RNMPlugin = new ReactNativeManualDevicePlugin();
const appInsights = new ApplicationInsights({   
    config: { 
        instrumentationKey: 'YOUR_INSTRUMENTATION_KEY_GOES_HERE',
        extensions: [RNMPlugin],
        extensionConfig: {
            AppInsightsReactNativePlugin: {
                disableDeviceCollection: true,    // <----- The configuration must be passed into the ApplicationInsights config
            }
        }
    }
});
appInsights.loadAppInsights()
```

**Comment by MSNev** [comment]
In the case of the first one, this is sort of expected because we "allow" (historical fallback) configurations for extensions to be defined at the root of the config event if not defined. The only solution is to cast as any to effectively ignore this TypeScript type check error. As we don't really want to start adding this to the core `IConfig` or `IConfiguration`, if defined there it (should) work.

On the 2nd, this sounds like your build environment is still trying to drag in the previous (non-manual) implementation which does reference the device info. @siyuniu-ms can you please help provide any additional insights into how to stop Expo for trying to use the device info.

**Comment by poikjo** [comment]
Anything on this one? @siyuniu-ms

**Comment by siyuniu-ms** [comment]
Hi, I tried it again locally. I used 
```
import { ApplicationInsights } from '@microsoft/applicationinsights-web';
import { ReactNativeManualDevicePlugin } from '@microsoft/applicationinsights-react-native/manual';

var RNPlugin = new ReactNativeManualDevicePlugin();
var appInsights = new ApplicationInsights({
    config: {
        disableDeviceCollection: true,
        instrumentationKey: 'yourKeyHere',
        extensions: [RNPlugin]
    }
});
appInsights.loadAppInsights();
appInsights.trackPageView();
```
For expo web mode, this does not throw any build error and could send out telemetry.
I did spot the device-info install problem for android app though. It may be caused by expo-android specific settings. I am continue working on it.

**Comment by siyuniu-ms** [comment]
I've changed the export path in pr #29. It works both for android and web locally this time. I will upload an example also. Hope this could help.

**Comment by poikjo** [comment]
Got my project working with your sample code so on my side this looks fine. Thanks a lot!

**Comment by MSNev** [comment]
> I've changed the export path in pr https://github.com/microsoft/applicationinsights-react-native/pull/29. It works both for android and web locally this time. I also upload an example https://github.com/siyuniu-ms/expo-manual-demo. Hope this could help.

We should add this as an example in this repo.

**Comment by siyuniu-ms** [comment]
> > I've changed the export path in pr #29. It works both for android and web locally this time. I also upload an example https://github.com/siyuniu-ms/expo-manual-demo. Hope this could help.
> 
> We should add this as an example in this repo.

Thanks, the example is now included in #29
